### PR TITLE
Next

### PR DESCRIPTION
**Pull request by franz**
Fixes a bunch of issues for 1.0/1.2 milestones

### CODE REVIEW

File: config.h.in.cmake [hunk 1]
```diff
@@ -151,6 +151,9 @@
 /* Disable cl_khr_fp16 because fp16 is not supported */
 #cmakedefine _CL_DISABLE_HALF
 
+/* Disable cl_khr_fp64 because fp64 is not supported */
```
Comment (pjaaskel) [inline config.h.in.cmake]: Does it disable it for all devices? What about heterogeneous platforms where host does not support fp64 but the co-processor (GPU) does?

Comment (franz) [inline config.h.in.cmake]: No, this variable is only for host CPU (perhaps i should rename it) and only used in pthread/basic drivers. 

The other drivers each have their own supported extension lists in CMake (used @ building the kernel library). In driver code, CUDA and HSA currently  hardcode double support to fully supported, TCE hardcodes it to `CL_FP_ROUND_TO_NEAREST | CL_FP_INF_NAN`

Comment (franz) [inline config.h.in.cmake]: _CL_DISABLE_HALF is also detected on CPU only.

**Comment by inducer** [comment]
Confirmed fixed. Thanks for working on this!

**Pull request merged**